### PR TITLE
bond: Add support for the ports-config

### PR DESCRIPTION
**Pull request by liangwen12year**
The bond port priority is only supported to set when bonding mode is active-backup(1) or balance-tlb (5) or balance-alb (6).

### CODE REVIEW

**Comment by liangwen12year** [comment]
This PR is still in draft status.

File: examples/bond_port_up.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  - name: bond0
+    type: bond
+    state: up
+    link-aggregation:
+      mode: active-backup
+      options:
+        miimon: 140
+      port:
+        - eth1
+        - eth2
```
Comment (cathay4t) [inline examples/bond_port_up.yml:14]: We should not suggest user to define port names in two places.

And in commit message, we should explain an error will happen if `port` does not agree with `ports-config`.

File: rust/src/lib/ifaces/bond.rs [hunk 2]
```diff
@@ -104,8 +105,17 @@ impl BondInterface {
         }
     }
 
+    fn sort_ports_config(&mut self) {
+        if let Some(ref mut bond_conf) = self.bond {
+            if let Some(ref mut port_conf) = &mut bond_conf.ports_config {
+                port_conf.sort_unstable_by_key(|p| p.name.clone())
```
Comment (cathay4t) [inline rust/src/lib/ifaces/bond.rs:111]: Just a rust quiz: could you explain why we need `clone()` here?

File: rust/src/python/libnmstate/schema.py [hunk 3]
```diff
@@ -169,6 +169,12 @@ class Bond:
     PORT = "port"
     PORTS = "ports"
     OPTIONS_SUBTREE = "options"
+    PORTS_CONFIG_SUBTREE = "ports-config"
+
+    class PortsConfig:
+        NAME = "name"
+        PRIO = "prio"
```
Comment (cathay4t) [inline rust/src/python/libnmstate/schema.py]: Please use `PRIORITY = "priority"`.

File: rust/src/lib/ifaces/bond.rs [hunk 4]
```diff
@@ -1282,3 +1346,46 @@ impl MergedInterface {
...
+#[non_exhaustive]
+pub struct BondPortConfig {
+    /// name is mandatory when specifying the ports configuration.
+    pub name: String,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        default,
+        deserialize_with = "crate::deserializer::option_i32_or_string"
+    )]
+    pub prio: Option<i32>,
```
Comment (cathay4t) [inline rust/src/lib/ifaces/bond.rs]: Please use `priority` instead of `prio`. This user facing, abbreviation is bad idea.

Comment (cathay4t) [reply rust/src/lib/ifaces/bond.rs]: And document please.

**Review by cathay4t — CHANGES_REQUESTED** [review]
* Change property name from `prio` to `priority`
* Add unit test case for conflict YAML between `port` and `ports-config`.

File: rust/src/lib/ifaces/bond.rs (hunk 2 above)
Comment (liangwen12year) [reply rust/src/lib/ifaces/bond.rs:111]: It relates to Rust's ownership, which prevents moving the element in a collection during sorting unless the elements are cloned ?

File: examples/bond_port_up.yml (hunk 1 above)
Comment (liangwen12year) [reply examples/bond_port_up.yml:14]: I revised the commit message and added the unit case to assert that an error will be raised.

Comment (liangwen12year) [reply examples/bond_port_up.yml:14]: Maybe I need to add some validation function to raise specific errors.

**Comment by liangwen12year** [comment]
@cathay4t ,thanks for reviewing, I think that all your comments are addressed now.

**Review by cathay4t — APPROVED** [review]
LGTM

**Pull request merged**